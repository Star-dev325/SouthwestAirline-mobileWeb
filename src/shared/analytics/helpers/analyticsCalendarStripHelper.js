// @flow
import dayjs from 'dayjs';

import type { BoundType } from 'src/shared/flow-typed/flightBound.types';
import { INBOUND, OUTBOUND } from 'src/shared/constants/flightBoundTypes';

export const calculateDateDiffs = (
  newDate: string,
  previousDate: string,
  direction: BoundType,
  currentDiffs: { [BoundType]: string } = {}
): { [BoundType]: string } => {
  const diff = dayjs(newDate).diff(previousDate, 'days');
  const sign = Math.sign(diff) === 1 ? '+' : '';
  const diffs = { ...currentDiffs };

  diffs[direction] = `${sign}${diff}`;

  return diffs;
};

export const calculateOverrideDateDiffs = (
  newDate: string,
  previousDate: string,
  direction: BoundType,
  isOverrideEndDate: boolean,
  previousDiffs: { [BoundType]: string } = {}
): { [BoundType]: string } => {
  let diffs = {};

  if (direction === OUTBOUND) {
    if (isOverrideEndDate) {
      diffs = calculateDateDiffs(newDate, previousDate, INBOUND, previousDiffs);
    } else {
      diffs = {};
    }
  }

  return diffs;
};
