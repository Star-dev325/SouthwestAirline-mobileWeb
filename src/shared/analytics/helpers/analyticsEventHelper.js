import _ from 'lodash';

const pageIdentifierTransformer = [
  { matcher: /^\/$|^\/\?/, id: 'home' },
  { matcher: /\/flight-status\/(?!.*html).*/, id: 'flight-status-results' },

  { matcher: /\/flight-status\?.*_modal=originAirport/, id: 'flight-status-from' },
  { matcher: /\/flight-status\?.*_modal=destinationAirport/, id: 'flight-status-to' },

  { matcher: /\/air\/booking\/shopping\?.*_modal=from/, id: 'air-booking-shopping-from' },
  { matcher: /\/air\/booking\/shopping\?.*_modal=to/, id: 'air-booking-shopping-to' },
  { matcher: /\/air\/booking\/shopping\?.*_modal=airBookingCalendar/, id: 'air-booking-shopping-dates' },

  { matcher: /\/air\/change\/shopping\?.*_modal=from/, id: 'air-change-shopping-from' },
  { matcher: /\/air\/change\/shopping\?.*_modal=to/, id: 'air-change-shopping-to' },
  { matcher: /\/air\/change\/shopping\?.*_modal=airBookingCalendar/, id: 'air-change-shopping-dates' },

  { matcher: /\/car\/booking\?.*_modal=pickUpModal/, id: 'car-booking-pick-up' },
  { matcher: /\/car\/booking\?.*_modal=dropOffModal/, id: 'car-booking-drop-off' },
  { matcher: /\/car\/booking\?.*_modal=carBookingCalendar/, id: 'car-booking-dates' },

  { matcher: /\/check-in\/reservation\/\w{6}$/, id: 'checkin-view-reservation' }
];

const _matchPath = (path) =>
  _.get(
    _.find(pageIdentifierTransformer, (t) => path.match(t.matcher)),
    'id'
  );

export const transformPath = (path, search = null) => {
  if (!path) {
    return '';
  }
  let toMatchPath = path;

  if (search) {
    toMatchPath = `${path}${search}`;
  }

  return _matchPath(toMatchPath) || path.replace(/\//g, '-').slice(1);
};

export const raiseEvent = (eventName) => {
  window._trackAnalytics && window._trackAnalytics.event();

  const customEvent = new window.CustomEvent(eventName);

  document.body.dispatchEvent(customEvent);
};

export const raiseSatelliteEvent = (eventName, detail) => {
  if (window._satellite) {
    detail ? window._satellite.track(eventName, detail) : window._satellite.track(eventName);
  }
};

export const customerMessageAnalytics = (messages) => {
  if (messages) {
    const messageKeys = _.chain(messages).map('key').compact().join().value();

    _.set(window, 'data_a.message.customer', messageKeys);
    _.set(window, 'data_a.message.customerdisplay', '1');
  }
};

export const getPageIdentifier = (location) => {
  const { pathname = '', search = '' } = location || {};

  return transformPath(pathname, search);
};
