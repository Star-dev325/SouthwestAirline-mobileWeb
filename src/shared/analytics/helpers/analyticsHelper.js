import _ from 'lodash';

export const generateUpdatedFlowStoreForAnalytics = (flowSelectors, state, actionType) =>
  _.chain(flowSelectors)
    .pickBy((selectorObj) => _.includes(selectorObj.actions, actionType))
    .mapValues((selectorObj) => selectorObj.selector(state))
    .value();

export const generateFlowActionListForAnalytics = (flowSelectors) =>
  _.chain(flowSelectors).mapValues('actions').values().flatten().uniq().value();

export const generateUpdatedStoresForAnalytics = (generators, state, actionType) =>
  _.reduce(
    generators,
    (result, generator, key) => {
      _.each(generator(state, actionType), (subValue, subKey) => {
        result[`${key}.${subKey}`] = subValue;
      });

      return result;
    },
    {}
  );
