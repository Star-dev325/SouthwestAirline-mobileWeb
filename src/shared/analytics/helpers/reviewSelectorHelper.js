import _ from 'lodash';
import { transformToUnitPrice } from 'src/airBooking/transformers/transformToEarlyBirdPriceDetails';

export const buildReviewAnalyticsData = (
  earlyBirdEligibility,
  purposeOfTravel,
  earlyBirdInPathRadioButtonChecked,
  flightPricingPageAnalyticsData,
  earlyBirdSelected
) => ({
  earlyBirdBasePriceCents: _.get(earlyBirdEligibility, 'unitPrice.amount', null),
  earlyBirdEstimatedTotalCents: _.get(earlyBirdEligibility, 'totalPrice.amount', null),
  earlyBirdOptionSelected: earlyBirdInPathRadioButtonChecked || earlyBirdSelected,
  earlyBirdOptionShown: _.some(_.get(earlyBirdEligibility, 'bounds', []), 'isEligible'),
  purposeOfTravel,
  earlyBirdBasePriceOutbound: _.get(transformToUnitPrice(_.get(earlyBirdEligibility, 'bounds[0]')), 'amount', null),
  earlyBirdBasePriceInbound: _.get(transformToUnitPrice(_.get(earlyBirdEligibility, 'bounds[1]')), 'amount', null),
  ...flightPricingPageAnalyticsData
});
