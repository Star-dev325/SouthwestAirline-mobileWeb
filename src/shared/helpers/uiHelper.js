export const scrollToTop = () => {
  const appDom = document.querySelector('.app__contents');

  if (appDom) {
    setTimeout(() => {
      appDom.scrollTop = 0;
    }, 0);
  }
};

export const forceDomUpdatesAndThenExecute = (callbackFn) => {
  requestAnimationFrame(() => {
    requestAnimationFrame(() => callbackFn());
  });
};
