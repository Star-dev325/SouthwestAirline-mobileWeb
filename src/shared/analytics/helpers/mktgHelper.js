// @flow
import _ from 'lodash';
import mktgDataConstants from 'src/shared/constants/mktgDataConstants';
import { createSelector } from 'src/shared/helpers/createSelectorWithErrorReporter';
import { getGlobalMktgData } from 'src/shared/analytics/selectors/getGlobalMktgData';
import { toNumberStringWithoutCommas } from 'src/shared/helpers/currencyValueHelper';
import type { FlightProductCard, ProductDefinition } from 'src/shared/flow-typed/shared.types';
import type { MKTGFareType } from 'src/shared/flow-typed/mktg.types';

const {
  NOT_AVAILABLE,
  NOT_SELECTED,
  DIRECT_FLIGHT,
  POINTS_BOOKING,
  CURRENCY_BOOKING,
  CONNECTING_FLIGHT,
  NONSTOP_FLIGHT
} = mktgDataConstants;

export const parseFareValue = (fare: MKTGFareType, fareValue: (fare: MKTGFareType) => string) =>
  (typeof fare === 'string' && fare) || (fare.reasonIfUnavailable && NOT_AVAILABLE) || fareValue(fare);

export const getFarePrice = (fare: MKTGFareType) => {
  const pricePath = fare.discountedPrice ? 'discountedPrice.amount' : 'price.amount';
  const farePrice = _.get(fare, pricePath, NOT_AVAILABLE);

  return (
    (farePrice === NOT_AVAILABLE && farePrice) ||
    (parseInt(farePrice, 10) === 0 && 'zero') ||
    toNumberStringWithoutCommas(farePrice)
  );
};

export const getFarePriceDifference = ({ priceDifference: { amount = NOT_AVAILABLE, sign = '' } = {} }: MKTGFareType) =>
  (amount === NOT_AVAILABLE && amount) ||
  (parseInt(amount) === 0 && 'zero') ||
  (sign === '-' && `${sign}${toNumberStringWithoutCommas(amount)}`) ||
  toNumberStringWithoutCommas(amount);

export const getSortedFareProducts = (products: Array<ProductDefinition>): Array<ProductDefinition> =>
  [...products].sort((product1, product2) => (product2.rowOrder > product1.rowOrder ? 1 : -1));

export const getSortedFares = (fares: Array<MKTGFareType>, products: Array<ProductDefinition>): Array<MKTGFareType> =>
  products.map(
    ({ productId }) =>
      fares.find((fare) => typeof fare !== 'string' && fare._meta.fareProductId === productId) || NOT_AVAILABLE
  );

export const getFareData = (
  propPrefix: string,
  fares: Array<MKTGFareType>,
  products: Array<ProductDefinition>,
  bookingType: string,
  fareValue: (MKTGFareType) => string = getFarePrice
) =>
  products.reduce((fareData, product, index) => {
    const fare = fares.find(
      (fareItem) => typeof fareItem !== 'string' && fareItem._meta.fareProductId === product.productId
    );
    const {
      [`${propPrefix}_viewfaretypes`]: fareTypes,
      [`${propPrefix}_viewfareproductids`]: fareIds,
      [`${propPrefix}_viewfarecurrency`]: fareCurrencyList,
      [`${propPrefix}_viewfarepoints`]: farePointsList
    } = fareData;
    const fareValueResult =
      (fare && parseFareValue(fare, fareValue)) ||
      (fares.every((fareItem) => fareItem === NOT_SELECTED) && NOT_SELECTED) ||
      NOT_AVAILABLE;
    const fareCurrency = bookingType === CURRENCY_BOOKING ? fareValueResult : NOT_SELECTED;
    const farePoints = bookingType === POINTS_BOOKING ? fareValueResult : NOT_SELECTED;
    const { id: fareType = NOT_AVAILABLE, productId: fareProductId = NOT_AVAILABLE } = product;
    const dataIndex = index + 1;

    return {
      ...fareData,
      [`${propPrefix}_viewfaretypes`]: fareTypes ? `${fareTypes}|${fareType}` : fareType,
      [`${propPrefix}_viewfareproductids`]: fareIds ? `${fareIds}|${fareProductId}` : fareProductId,
      [`${propPrefix}_viewfarecurrency`]: fareCurrencyList ? `${fareCurrencyList}|${fareCurrency}` : fareCurrency,
      [`${propPrefix}_viewfarepoints`]: farePointsList ? `${farePointsList}|${farePoints}` : farePoints,
      [`${propPrefix}_viewfareproductid${dataIndex}`]: fareType,
      [`${propPrefix}_viewfaretype${dataIndex}`]: fareType,
      [`${propPrefix}_viewfarecurrency${dataIndex}`]: fareCurrency,
      [`${propPrefix}_viewfarepoints${dataIndex}`]: farePoints
    };
  }, {});

export const getFareDifferencesData = (propPrefix1: string, propPrefix2: string, fareDifferences: Array<string>) =>
  fareDifferences.reduce(
    (result, difference, index) => ({
      ...result,
      [`${propPrefix1}_viewfarediff${propPrefix2}${index + 1}`]: difference
    }),
    {}
  );

export const getStopData = (propPrefix: string, flight: FlightProductCard) => {
  const numberOfStops = _.get(flight, '_meta.numberOfStops', NOT_SELECTED);
  const hasStop = parseInt(numberOfStops) > 0;
  const hasPlaneChange = !!flight.stopCity;

  const stopType = hasPlaneChange ? CONNECTING_FLIGHT : hasStop ? DIRECT_FLIGHT : NONSTOP_FLIGHT;

  return {
    [`${propPrefix}_stops`]: numberOfStops,
    [`${propPrefix}_stoptype`]: numberOfStops === NOT_SELECTED ? NOT_SELECTED : stopType
  };
};

export const getMktgDataForPage = (path: string, defaultValue: *) => (state: *) => _.get(state, path, defaultValue || {});

export const createMktgDataSelector = (path: string, defaultValue: *) => createSelector(
  [getGlobalMktgData, getMktgDataForPage(path, defaultValue)],
  (globalMktgData, pageMktgData) => ({ ...globalMktgData, ...pageMktgData })
);
