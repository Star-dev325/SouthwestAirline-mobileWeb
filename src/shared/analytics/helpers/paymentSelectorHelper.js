import _ from 'lodash';

import { getCardType } from 'src/shared/helpers/creditCardHelper';

import {
  NEW_CREDIT_CARD_ID,
  RAPID_REWARDS_VISA_ID,
  PAY_PAL_CARD_ID,
  APPLE_PAY_CARD_ID,
  UPLIFT_CARD_ID
} from 'src/shared/constants/creditCardConstants';
import { INSTANT_CREDIT_RAPID_REWARDS_VISA, PAYPAL, APPLE_PAY, UATP } from 'src/shared/constants/creditCardTypes';
import { addCurrency } from 'src/shared/api/helpers/currencyHelper';
import { isCurrencyAmountZero } from 'src/shared/helpers/travelFundsHelper';

export const buildPaymentAnalyticsData = (
  paymentInfo,
  savedCreditCards,
  flightConfirmationPage,
  travelFundsBalanceRemaining
) => {
  let typeOfCreditCard;
  const { city, stateProvinceRegion, zipOrPostalCode, saveCreditCard, cardNumber } = paymentInfo;
  const fundsApplied = _.get(flightConfirmationPage, 'fundsApplied');
  let selectedCardId = _.get(paymentInfo, 'selectedCardId');
  const selectedSavedCreditCard =
    _.find(_.chain(savedCreditCards).values().flatten().value(), { savedCreditCardId: selectedCardId }) || {};
  const creditCardType = getCardType(cardNumber);
  const primaryCardType = _.get(savedCreditCards, 'primaryCard.type');
  const rtfs = [];
  const luvVouchers = [];
  const giftCards = [];
  const isApplePay = _.isEqual(selectedCardId, APPLE_PAY_CARD_ID);
  const isUplift = _.isEqual(selectedCardId, UPLIFT_CARD_ID);

  if (!_.isEmpty(fundsApplied)) {
    let fundTypes = [];

    if (!_.isEmpty(paymentInfo) && !isCurrencyAmountZero(travelFundsBalanceRemaining)) {
      if (_.isEmpty(creditCardType) && selectedSavedCreditCard.type) {
        fundTypes = [selectedSavedCreditCard.type];
      } else if (isApplePay) {
        fundTypes = [APPLE_PAY.shortName];
      } else if (isUplift) {
        fundTypes = [UATP.shortName];
      } else if (_.isEmpty(creditCardType)) {
        fundTypes = [INSTANT_CREDIT_RAPID_REWARDS_VISA.key];
      } else {
        fundTypes = [creditCardType];
      }
    } else if (_.isEmpty(paymentInfo) && !isCurrencyAmountZero(travelFundsBalanceRemaining)) {
      fundTypes = [primaryCardType];
    }

    _.forEach(fundsApplied, (fund) => {
      const fundType = _.get(fund, 'travelFundType');
      const appliedAmount = _.get(fund, 'appliedAmount');

      !_.includes(fundTypes, fundType) && fundTypes.push(fundType);
      fundType === 'TRAVEL_FUNDS' && rtfs.push(appliedAmount);
      fundType === 'LUV_VOUCHER' && luvVouchers.push(appliedAmount);
      fundType === 'GIFT_CARD' && giftCards.push(appliedAmount);

      typeOfCreditCard = fundTypes.join(',');
    });
  } else {
    if (_.isEmpty(paymentInfo)) {
      typeOfCreditCard = isApplePay ? APPLE_PAY.shortName : primaryCardType;
      typeOfCreditCard = isUplift ? UATP.shortName : typeOfCreditCard;
      selectedCardId = _.get(savedCreditCards, 'primaryCard.savedCreditCardId');
    } else {
      typeOfCreditCard = _.isEmpty(creditCardType)
        ? selectedSavedCreditCard.type || INSTANT_CREDIT_RAPID_REWARDS_VISA.key
        : creditCardType;

      typeOfCreditCard = isApplePay ? APPLE_PAY.shortName : typeOfCreditCard;
      typeOfCreditCard = isUplift ? UATP.shortName : typeOfCreditCard;
    }
  }

  const type = selectedCardId === PAY_PAL_CARD_ID ? PAYPAL.key : typeOfCreditCard;

  return _.omitBy(
    {
      city,
      state: stateProvinceRegion,
      zipcode: zipOrPostalCode,
      saveNewCardSelected: _.toBoolean(saveCreditCard) || false,
      storedCard:
        selectedCardId &&
        !isCurrencyAmountZero(travelFundsBalanceRemaining) &&
        !_.includes(
          [NEW_CREDIT_CARD_ID, RAPID_REWARDS_VISA_ID, PAY_PAL_CARD_ID, APPLE_PAY_CARD_ID, UPLIFT_CARD_ID],
          selectedCardId
        ),
      type,
      travelfund:
        rtfs.length > 1
          ? _.get(addCurrency(...rtfs), 'amount')
          : !_.isEmpty(rtfs[0])
            ? _.get(rtfs[0], 'amount')
            : undefined,
      luvvoucher:
        luvVouchers.length > 1
          ? _.get(addCurrency(...luvVouchers), 'amount')
          : !_.isEmpty(luvVouchers[0])
            ? _.get(luvVouchers[0], 'amount')
            : undefined,
      giftcard:
        giftCards.length > 1
          ? _.get(addCurrency(...giftCards), 'amount')
          : !_.isEmpty(giftCards[0])
            ? _.get(giftCards[0], 'amount')
            : undefined
    },
    _.isUndefined
  );
};
