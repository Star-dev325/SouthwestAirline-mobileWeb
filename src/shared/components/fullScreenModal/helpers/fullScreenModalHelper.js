import { history } from 'src/appHistory';
import BrowserObject from 'src/shared/helpers/browserObject';
import { mergeQuery, combineUri, getQueryStringParameterByKey } from 'src/shared/helpers/urlHelper';
import { addEventListenerOnce } from 'src/shared/helpers/eventHelpers';
import Q from 'q';

const { location } = BrowserObject;

export const showFullScreenModal = (id) => {
  if (getQueryStringParameterByKey('_modal') !== id) {
    return Q(history.push(combineUri(location.pathname, mergeQuery({ _modal: id }, location.search))));
  } else {
    return Q();
  }
};

export const hideFullScreenModal = (id) => {
  if (getQueryStringParameterByKey('_modal') === id) {
    const deferred = Q.defer();

    addEventListenerOnce(window, 'popstate', () => {
      deferred.resolve();
    });
    history.goBack();

    return deferred.promise;
  } else {
    return Q();
  }
};

export const hideModalAndUpdateFormField = (id, fieldName, fieldValue, onChange) => {
  hideFullScreenModal(id);
  onChange(fieldName, fieldValue);
};

export const getModalId = () => getQueryStringParameterByKey('_modal');
