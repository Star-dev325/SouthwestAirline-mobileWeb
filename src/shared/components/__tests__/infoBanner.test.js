import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import React from 'react';
import InfoBanner from 'src/shared/components/infoBanner';
import * as uiHelper from 'src/shared/helpers/uiHelper';

describe('InfoBanner', () => {
  const scrollToTopMock = jest.spyOn(uiHelper, 'scrollToTop');
  const componentDidUpdateProps = {
    body: 'new body',
    header: 'newHeader'
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render banner with header and body', () => {
    const { container } = createComponent();

    expect(container).toMatchSnapshot();
  });

  it('should not call scrollToTop on componentDidUpdate if shouldScrollToTop is falsy', () => {
    createComponent(componentDidUpdateProps);

    expect(scrollToTopMock).not.toHaveBeenCalled();
  });

  it('should call scrollToTop on componentDidUpdate if shouldScrollToTop is set to true', () => {
    const instance = React.createRef();

    createComponent({
      body: 'new body',
      header: 'newHeader',
      shouldScrollToTop: true,
      ref: instance
    });

    instance.current.componentDidUpdate(instance.current.props);

    expect(scrollToTopMock).toHaveBeenCalled();
  });

  it('should render with className is one is passed in', () => {
    const { container } = createComponent({ className: 'test-classname' });

    expect(container).toMatchSnapshot();
  });

  it('should render learn more link if one is passed in', () => {
    const { container } = createComponent({ learnMoreUrl: 'mock-url' });

    expect(container).toMatchSnapshot();
  });

  it('should render icon that is passed in as iconType', () => {
    const { container } = createComponent({ iconType: 'check-circle' });

    expect(container).toMatchSnapshot();
  });

  it('should render icon color that is passed in', () => {
    const { container } = createComponent({ iconTypeColor: 'error' });

    expect(container).toMatchSnapshot();
  });

  const createComponent = (props = {}) => {
    const defaultProps = {
      header: 'This is a header',
      body: 'This is a body',
      learnMoreUrl: null,
      iconTypeColor: null
    };
    const finalProps = { ...defaultProps, ...props };

    return render(<InfoBanner {...finalProps} />);
  };
});
