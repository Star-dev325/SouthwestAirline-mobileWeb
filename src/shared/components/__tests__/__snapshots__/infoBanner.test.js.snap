// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoBanner should render banner with header and body 1`] = `
<div>
  <div
    class="info-banner "
  >
    <div
      class="info-banner--alert"
    >
      <i
        class="icon icon_exclamation-circle warning"
      />
    </div>
    <div
      class="info-banner-container"
    >
      <b
        class="info-banner-container--header"
      >
        This is a header
      </b>
      <p
        class="info-banner-container--body"
      >
        This is a body
      </p>
      <div
        class="info-banner-container--link"
      />
    </div>
  </div>
</div>
`;

exports[`InfoBanner should render icon color that is passed in 1`] = `
<div>
  <div
    class="info-banner "
  >
    <div
      class="info-banner--alert"
    >
      <i
        class="icon icon_exclamation-circle error"
      />
    </div>
    <div
      class="info-banner-container"
    >
      <b
        class="info-banner-container--header"
      >
        This is a header
      </b>
      <p
        class="info-banner-container--body"
      >
        This is a body
      </p>
      <div
        class="info-banner-container--link"
      />
    </div>
  </div>
</div>
`;

exports[`InfoBanner should render icon that is passed in as iconType 1`] = `
<div>
  <div
    class="info-banner "
  >
    <div
      class="info-banner--alert"
    >
      <i
        class="icon icon_check-circle"
      />
    </div>
    <div
      class="info-banner-container"
    >
      <b
        class="info-banner-container--header"
      >
        This is a header
      </b>
      <p
        class="info-banner-container--body"
      >
        This is a body
      </p>
      <div
        class="info-banner-container--link"
      />
    </div>
  </div>
</div>
`;

exports[`InfoBanner should render learn more link if one is passed in 1`] = `
<div>
  <div
    class="info-banner "
  >
    <div
      class="info-banner--alert"
    >
      <i
        class="icon icon_exclamation-circle warning"
      />
    </div>
    <div
      class="info-banner-container"
    >
      <b
        class="info-banner-container--header"
      >
        This is a header
      </b>
      <p
        class="info-banner-container--body"
      >
        This is a body
      </p>
      <div
        class="info-banner-container--link"
      >
        <a
          href="mock-url"
          target="_blank"
        >
          Learn More 
          <i
            class="pblue xxlarge learn-more--icon icon icon_keyboard-arrow-right"
          />
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`InfoBanner should render with className is one is passed in 1`] = `
<div>
  <div
    class="info-banner test-classname"
  >
    <div
      class="info-banner--alert"
    >
      <i
        class="icon icon_exclamation-circle warning"
      />
    </div>
    <div
      class="info-banner-container"
    >
      <b
        class="info-banner-container--header"
      >
        This is a header
      </b>
      <p
        class="info-banner-container--body"
      >
        This is a body
      </p>
      <div
        class="info-banner-container--link"
      />
    </div>
  </div>
</div>
`;
